[ci/relay] TestLocator_* flakes on Darwin runners (HELLO i/o timeout)

## Summary

Tests in \`relay/server/cluster\` that drive the \`Locator\` (multi-node cluster
on \`127.0.0.1\`) flake on the Darwin (macOS) GitHub Actions runner with the same
\`i/o timeout\` on the HELLO handshake. Linux / Windows / FreeBSD all pass on the
same SHA.

Affected tests (rotates between runs — same root cause):
- \`TestLocator_LookupCacheHitSkipsBroadcast\`
- \`TestLocator_ConcurrentLookupsShareWaiter\`
- \`TestLocator_LookupHitsRemote\`

## Symptom

\`\`\`
--- FAIL: TestLocator_LookupCacheHitSkipsBroadcast (0.20s)
time="…Z" level=warning msg="cluster: accept from 127.0.0.1:49446: read HELLO:
  read tcp 127.0.0.1:49445->127.0.0.1:49446: i/o timeout"
FAIL  github.com/openzro/openzro/relay/server/cluster  4.560s
\`\`\`

## Repro

- Tag-push runs on macOS-latest: e.g. https://github.com/openzro/openzro/actions/runs/26352698931
- PR runs on macOS-latest: e.g. https://github.com/openzro/openzro/actions/runs/26352224443
- Rerun almost always passes (confirmed multiple times across #110, #113, #114, alpha.80 tag)

## Root cause hypothesis

Locator tests spin up cluster nodes on loopback and require the HELLO
handshake to complete within a tight deadline. macOS runners (slower loopback,
shared / contended hosts) miss that deadline more often than Linux runners.
Pure timing flake — no functional regression in the relay code.

## Proposed fix

Bump the HELLO deadline (or read deadline) in the test setup of
\`relay/server/cluster/locator_test.go\` so the tests tolerate macOS-runner
loopback latency without becoming meaningfully slower on the Linux runner.

Out-of-scope alternative: skip the affected tests on Darwin via a build tag.
Less ideal — masks the surface without fixing it.

## Why this matters now

Recurring rerun cost on every PR + every tag push. Cheap to fix once;
expensive to keep eating.

## Not a blocker

Pre-existing. Phase 1/2/4 of #108 all merged with this same flake — the
rerun is documented in PR threads.

--- a/relay/server/cluster/forwarder_test.go
+++ b/relay/server/cluster/forwarder_test.go
@@ -3,8 +3,6 @@ package cluster
 import (
 	"context"
 	"errors"
-	"os"
-	"runtime"
 	"sync"
 	"testing"
 	"time"
@@ -109,23 +107,6 @@ func startForwarderPair(t *testing.T, ownsA, ownsB []messages.PeerID) (
 	string, string,
 ) {
 	t.Helper()
-
-	// Every cross-pod forwarder test goes through here, which Dials a
-	// bidirectional inter-pod TCP HELLO handshake on loopback. The
-	// accept side bounds the HELLO read by the production const
-	// helloTimeout (3s, transport.go) — not overridable from a test.
-	// The macOS CI runner is contended enough that a loopback HELLO
-	// read intermittently exceeds 3s, the connection is dropped, the
-	// peer never registers, and the test fails with "peer not
-	// connected anywhere" / read HELLO i/o timeout. This is a
-	// non-hermetic timing dependency on a slow shared runner, not a
-	// forwarder bug — the logic is OS-agnostic and fully covered on
-	// Linux CI + locally. Skip on darwin CI only (mirrors the
-	// TestServiceLifecycle FreeBSD-CI precedent); proper hardening of
-	// the handshake budget is tracked separately.
-	if runtime.GOOS == "darwin" && os.Getenv("CI") == "true" {
-		t.Skip("non-hermetic on macOS CI: contended runner exceeds the 3s loopback HELLO budget — covered on Linux")
-	}
 
 	dispA := newFakeDispatcher(ownsA...)
 	dispB := newFakeDispatcher(ownsB...)

--- a/relay/server/cluster/transport.go
+++ b/relay/server/cluster/transport.go
@@ -115,9 +115,18 @@ func (s *Stream) readLoop(handler FrameHandler) {
 
 // helloTimeout caps how long an accepted connection has to send
 // its HELLO frame before we drop it. Real intra-cluster handshakes
-// finish in microseconds; 3 s leaves plenty of room for slow CNI
-// startup without enabling a long-tail DoS via half-open conns.
-const helloTimeout = 3 * time.Second
+// finish in microseconds, but contended macOS CI runners have been
+// observed taking >3 s on loopback (issue #116). 10 s holds enough
+// headroom for the slowest hosts we've measured without enabling a
+// long-tail DoS via half-open conns — the relay's connection caps
+// and rate limits enforce the actual anti-abuse budget; this value
+// only governs how long ONE half-open conn is allowed to dangle.
+//
+// Declared as `var` (not `const`) so transport_test.go can shrink
+// it locally to keep TestTransport_AcceptedConnWithoutHelloTimesOut
+// fast — restored by t.Cleanup, no shared-state leakage between
+// tests in the package.
+var helloTimeout = 10 * time.Second
 
 // classifyHelloReject maps a DecodeHello failure to a metric label.
 // Pattern-matching on the error text keeps the cluster package

--- a/relay/server/cluster/transport_test.go
+++ b/relay/server/cluster/transport_test.go
@@ -187,6 +187,14 @@ func TestTransport_AcceptedConnWithoutHelloIsDropped(t *testing.T) {
 func TestTransport_AcceptedConnWithoutHelloTimesOut(t *testing.T) {
 	// A dialer that opens the conn and never sends anything must
 	// be reaped by the helloTimeout, not held forever.
+	//
+	// Shrink the production budget locally so the test waits ~400ms
+	// instead of ~10s; the path under test (deadline fires → conn
+	// dropped → no stream registered) is value-independent.
+	orig := helloTimeout
+	helloTimeout = 200 * time.Millisecond
+	t.Cleanup(func() { helloTimeout = orig })
+
 	srvHandler := newCaptureHandler()
 	srv, addr := startTransport(t, srvHandler)
 

